openssl/src/x509/mod.rs

```diff
@@ -405,6 +405,18 @@ impl X509Ref {
         unsafe { ffi::X509_issuer_name_hash(self.as_ptr()) as u32 }
     }
 
+    /// Returns the extension index for the nid, starting at lastpos, if it exists.
+    #[corresponds(X509_get_ext_by_NID)]
+    pub fn ext_by_nid(&self, nid: Nid, lastpos: c_int) -> Result<c_int, ErrorStack> {
```

Comment:
This seems like a very raw API. What's the use case?

Reply:
This is used as an alternative to x509_get_ext_d2i when you don’t need the actual extension value, just to see if it is present

Reply:
If there's desire to do something like this externally I think I'd like a cleaner API, and if this is only for a test I don't think it's worth adding to the public API.

Reply:
Are you looking for exposing x509_get_ext_d2i itself, so we just have one method to get an extension (which implicitly see's if it is present), or are you looking for this API to expose less of OpenSSL's internal structure in this API. I could modify this API to be higher level, and just return a boolean if the extension was detected, dropping lastpos